packages/flutter_tools/lib/src/ios/code_signing.dart

```diff
@@ -116,229 +159,787 @@ Future<Map<String, String>?> getCodeSigningIdentityDevelopmentTeamBuildSetting({
+        '--clear-ios-signing-settings"',
+      );
+      return;
+    }
+
+    final _CodeSigningStyle? style = await _selectSigningStyle();
+    if (style == null) {
+      _logger.printWarning(_codeSignSelectionCanceled);
+      return;
+    }
+
+    if (style == _CodeSigningStyle.automatic) {
+      final List<String> validCodeSigningIdentities = await _getSigningIdentities();
+      if (validCodeSigningIdentities.isEmpty) {
+        _logger.printError(noCertificatesInstruction, emphasis: true);
+        _logger.printWarning(_codeSignSelectionCanceled);
+        return;
+      }
+      final String? identity = await _selectSigningIdentity(validCodeSigningIdentities);
+      if (identity == null) {
+        _logger.printWarning(_codeSignSelectionCanceled);
+        return;
+      }
+      _saveCodeSignIdentity(identity);
+    } else if (style == _CodeSigningStyle.manual) {
+      final List<_ProvisioningProfile> validProvisioningProfiles = await _getProvisioningProfiles();
+      if (validProvisioningProfiles.isEmpty) {
+        _logger.printError(
+          'No provisioning profiles were found. To learn how to create or download '
+          'a provisioning profile, please see '
+          'https://developer.apple.com/help/account/manage-provisioning-profiles/create-a-development-provisioning-profile',
+          emphasis: true,
         );
+        _logger.printWarning(_codeSignSelectionCanceled);
+        return;
+      }
+      final _ProvisioningProfile? profile = await _selectProvisioningProfile(
+        validProvisioningProfiles,
+      );
+      if (profile == null) {
+        _logger.printWarning(_codeSignSelectionCanceled);
+        return;
       }
+      _saveProvisioningProfile(profile);
+    }
+  }
+
+  /// Prompt user to select a code-signing style (Automatic or Manual).
+  Future<_CodeSigningStyle?> _selectSigningStyle() async {
+    _logger.printStatus('Code Signing Styles:', emphasis: true);
+    _logger.printStatus(
+      '  This setting specifies the method used to acquire and locate signing '
+      'assets. Choose Automatic to let Xcode automatically create and update '
+      'profiles, app IDs, and certificates. Choose Manual to create and update '
+      'these yourself on the developer website.',
+    );
+    _logger.printStatus('[1]: ${_CodeSigningStyle.automatic.label} (recommended)');
+    _logger.printStatus('[2]: ${_CodeSigningStyle.manual.label}');
+    final String choice = await _terminal.promptForCharInput(
+      List<String>.generate(2, (int number) => '${number + 1}')..add('q'),
+      prompt: 'Select a signing style (or "q" to quit)',
+      defaultChoiceIndex: 0, // Just pressing enter chooses the first one.
+      logger: _logger,
+      displayAcceptedCharacters: false,
+    );
+    if (choice == '1') {
+      return _CodeSigningStyle.automatic;
+    } else if (choice == '2') {
+      return _CodeSigningStyle.manual;
+    }
+    return null;
```

Comment:
Consider using a switch expression here:

```suggestion
    return switch (choice) {
      '1' => _CodeSigningStyle.automatic,
      '2' => _CodeSigningStyle.manual,
      _ => null,
    };
```